lambda/storeUsageInTimestream.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { storeUsageInTimestream } from './storeUsageInTimestream.js'
+import assert from 'node:assert'
+
+void describe('storeUsageInTimestream()', () => {
+	void it('should return the timestamp from cache if no newer data usage', async () => {
+		const iccid = '89457387300000022734'
+		const dataUsage = {
+			'89457387300000022734': {
+				'2020-02-03T07:49:13.000Z': {
+					usedBytes: 1424843,
+					billId: '7c5f95c7-c0ff-352e-93e2-5370a6d37b42',
+				},
+				/*'2020-02-03T08:04:14.000Z': {
+					usedBytes: 1103,
+					billId: '0538f940-b382-339c-b2a3-19c788c385f6',
+				},*/
+			},
+		}
+		const storeTimestream = storeUsageInTimestream({
+			getSimDetailsFromCache: async () =>
+				Promise.resolve({
+					sim: {
+						usedBytes: 1,
+						totalBytes: 2,
+						ts: new Date('2020-02-03T07:49:14.000Z'),
+					},
+					historyTs: '2020-02-03T07:49:13.000Z',
+				}),
+			storeHistoricalData: async () => Promise.resolve({ success: true }),
+		})
+		const res = await storeTimestream(iccid, dataUsage)
+		assert.equal(res, '2020-02-03T07:49:13.000Z')
+	})
+	void it('should return the timestamp of the newest usage from dataUsage', async () => {
+		const iccid = '89457387300000022734'
+		const dataUsage = {
+			'89457387300000022734': {
+				'2020-02-03T07:49:13.000Z': {
+					usedBytes: 1424843,
+					billId: '7c5f95c7-c0ff-352e-93e2-5370a6d37b42',
+				},
+				'2020-02-03T08:04:14.000Z': {
+					usedBytes: 1103,
+					billId: '0538f940-b382-339c-b2a3-19c788c385f6',
+				},
+			},
+		}
+		const storeTimestream = storeUsageInTimestream({
+			getSimDetailsFromCache: async () =>
+				Promise.resolve({
+					sim: {
+						usedBytes: 1,
+						totalBytes: 2,
+						ts: new Date('2020-02-03T07:49:14.000Z'),
+					},
+					historyTs: '2020-02-03T07:49:13.000Z',
+				}),
+			storeHistoricalData: async () => Promise.resolve({ success: true }),
+		})
+		const res = await storeTimestream(iccid, dataUsage)
+		assert.equal(res, '2020-02-03T08:04:14.000Z')
+	})
+	void it('should return correct timestamp if data storing fails', async () => {
+		const iccid = '89457387300000022734'
+		const dataUsage = {
+			'89457387300000022734': {
+				'2020-02-03T07:49:13.000Z': {
+					usedBytes: 1424843,
+					billId: '7c5f95c7-c0ff-352e-93e2-5370a6d37b42',
+				},
+				'2020-02-03T08:04:14.000Z': {
+					usedBytes: 1103,
+					billId: '0538f940-b382-339c-b2a3-19c788c385f6',
+				},
+			},
+		}
+		const storeTimestream = storeUsageInTimestream({
+			getSimDetailsFromCache: async () =>
+				Promise.resolve({
+					sim: {
+						usedBytes: 1,
+						totalBytes: 2,
+						ts: new Date('2020-02-03T07:49:14.000Z'),
+					},
+					historyTs: '2020-02-03T07:49:13.000Z',
+				}),
+			storeHistoricalData: async () =>
+				Promise.resolve({ error: new Error('Failed') }),
+		})
+		const res = await storeTimestream(iccid, dataUsage)
+		//What is actually the correct ts here? Should we use the 'old' one or the newest?
+		const expectedResult = '2020-02-03T07:49:13.000Z'
```

Comment:
Why does this function have to return a value at all?

Reply:
It seems not be used at all.

Reply:
It is used in 'getAllSimUsageOnomondo.ts' & 'storeSimInformationOnomondo.ts'. I need to store it in DynamoDB to know the last entry in Timestream for each Onomondo SIM. (This is since I'm getting all the data usage for a day when I fetch from Onomondo API and I want to write only the new ones to Timestream.)

Reply:
1. Why should storing fail? 
2. I don't think it should be `storeUsageInTimestream`'s task to return a value used in another function. You can take the date directly from the usage data.

Reply:
Good question, I was just thinking to have a solution in case it fails. I have now separated the different tasks in the function and refactored a bit as well as added tests to the new functions. Is this a better solution? 

https://github.com/bifravst/sim-details/pull/45/commits/2e6926e033e15f76c6dd2f07fca2909c37d84f89